src/controllers/confirm.company.controller.ts

```diff
@@ -27,39 +30,58 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const session: Session = req.session as Session;
     const companyNumber = req.query.companyNumber as string;
+
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    logger.debug("languageEnabled: " + locales.enabled + " for " + lang + " from " + locales.localesFolder);
+    logger.debug("localisations: " + JSON.stringify(locales.i18nCh.resolveNamespacesKeys(lang)));
+    
     const companyProfile: CompanyProfile = await getCompanyProfile(companyNumber);
 
-    const pageOptions = await buildPageOptions(session, companyProfile);
+    const pageOptions = await buildPageOptions(session, companyProfile, locales, lang);
     return res.render(Templates.CONFIRM_COMPANY, pageOptions);
     
   } catch (e) {
     return next(e);
   }
 };
 
-const buildPageOptions = async (session: Session, companyProfile: CompanyProfile): Promise<Object> => {
-  companyProfile = formatForDisplay(companyProfile);
+const buildPageOptions = async (session: Session, companyProfile: CompanyProfile, locales: LocalesService, lang: string): Promise<Object> => {
+  companyProfile = formatForDisplay(companyProfile, locales, lang);
   var addressArray: string[] = [companyProfile.registeredOfficeAddress.poBox,
     companyProfile.registeredOfficeAddress.premises, companyProfile.registeredOfficeAddress.addressLineOne,
     companyProfile.registeredOfficeAddress.addressLineTwo, companyProfile.registeredOfficeAddress.locality,
     companyProfile.registeredOfficeAddress.region, companyProfile.registeredOfficeAddress.country,
     companyProfile.registeredOfficeAddress.postalCode]
   const address = buildAddress(addressArray);
+ 
+  let backLinkUrl = COMPANY_LOOKUP.replace("{","%7B").replace("}","%7D");
+  if (lang != undefined) {
+    backLinkUrl += "%26lang%3D" + selectLang(lang);
+  }
   return {
+    ...getLocaleInfo(locales, lang),
+    currentUrl: Templates.CONFIRM_COMPANY + "?companyNumber=" + companyProfile.companyNumber,
+
     company: companyProfile,
+    notDueWarning: undefined,
     address: address,
     templateName: Templates.CONFIRM_COMPANY,
-    backLinkUrl: COMPANY_LOOKUP.replace("{","%7B").replace("}","%7D")
+    backLinkUrl
   };
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
+    logger.debug("POST confirm company " + req.url + " lang in query params: " + req.query.lang + ", and body: " + req.body.lang);
+
     const session: Session = req.session as Session;
+    const lang = req.body.lang ? selectLang(req.body.lang) : selectLang(req.query.lang);
```

Comment:
are we going to use `req.body.lang`? I am assuming the lang is always going to be a query param.